src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -131,57 +135,51 @@ public IReadOnlyNetworkPeerCollection ConnectedNodes
             IDateTimeProvider dateTimeProvider,
             INetworkPeerFactory networkPeerFactory)
         {
+            this.asyncLoopFactory = asyncLoopFactory;
+            this.connectedNodes = new NetworkPeerCollection();
             this.dateTimeProvider = dateTimeProvider;
+            this.loggerFactory = loggerFactory;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.Network = network;
+            this.networkPeerFactory = networkPeerFactory;
             this.NodeSettings = nodeSettings;
             this.nodeLifetime = nodeLifetime;
+            this.peerAddressManager = peerAddressManager;
+            this.Servers = new List<NetworkPeerServer>();
+
             this.Parameters = parameters;
             this.Parameters.ConnectCancellation = this.nodeLifetime.ApplicationStopping;
+            this.Parameters.UserAgent = $"{this.NodeSettings.Agent}:{this.GetVersion()}";
+            this.Parameters.Version = this.NodeSettings.ProtocolVersion;
 
-            this.loggerFactory = loggerFactory;
-            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
-
-            this.asyncLoopFactory = asyncLoopFactory;
+            // Don't start the peer discovery connector if the -connect arg is specified at startup.
+            if (!this.NodeSettings.ConnectionManager.Connect.Any())
```

Comment:
Generally I would prefer to keep logic in the Initialize method.
But as this is a very simple logic then I will let you decide.